pkg/tests/ossm/smcp_annotation_test.go

```diff
@@ -69,29 +68,43 @@ func TestSMCPAnnotations(t *testing.T) {
 			oc.WaitDeploymentRolloutComplete(t, ns, "testenv")
 
 			t.LogStep("Get annotations and verify that the pod has the expected: test1.annotation-from-smcp : test1, test2.annotation-from-smcp : [\"test2\"], test3.annotation-from-smcp : {test3}")
-			annotations := GetPodAnnotations(t, pod.MatchingSelector("app=env", ns))
-			assertAnnotationIsPresent(t, annotations, "test1.annotation-from-smcp", "test1")
-			assertAnnotationIsPresent(t, annotations, "test2.annotation-from-smcp", `["test2"]`)
-			assertAnnotationIsPresent(t, annotations, "test3.annotation-from-smcp", "{test3}")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				annotations := VerifyAndGetPodAnnotation(t, pod.MatchingSelector("app=env", ns))
+				assertAnnotationIsPresent(t, pod.MatchingSelector("app=env", ns), annotations, "test1.annotation-from-smcp", "test1")
+				assertAnnotationIsPresent(t, pod.MatchingSelector("app=env", ns), annotations, "test2.annotation-from-smcp", `["test2"]`)
+				assertAnnotationIsPresent(t, pod.MatchingSelector("app=env", ns), annotations, "test3.annotation-from-smcp", "{test3}")
+			})
 		})
 	})
 }
 
-func GetPodAnnotations(t TestHelper, podLocator oc.PodLocatorFunc) map[string]string {
-	annotations := map[string]string{}
+func VerifyAndGetPodAnnotation(t test.TestHelper, podLocator oc.PodLocatorFunc) map[string]string {
+	var data map[string]interface{}
 	po := podLocator(t, oc.DefaultOC)
-	retry.UntilSuccess(t, func(t test.TestHelper) {
-		output := shell.Executef(t, "kubectl get pod %s -n %s -o jsonpath='{.metadata.annotations}'", po.Name, po.Namespace)
-		err := json.Unmarshal([]byte(output), &annotations)
-		if err != nil {
-			t.Fatalf("Error parsing pod annotations json: %v", err)
-		}
-	})
+	yamlString := oc.GetYaml(t, po.Namespace, "pod", po.Name)
+	err := yaml.Unmarshal([]byte(yamlString), &data)
+	if err != nil {
+		t.Fatalf("Failed to unmarshal yaml: %s", err)
+	}
+
+	annotationsMap := data["metadata"].(map[interface{}]interface{})["annotations"].(map[interface{}]interface{})
```

Comment:
Oh, and I never wanted you to use `oc.GetYaml()` the way you use it here. Clearly, this is ugly. You can get the annotations without resorting to ugly maps like this.

Reply:
Fixed